hsu-test/main.rs

```diff
@@ -0,0 +1,90 @@
+use clap::{App, Arg};
+use std::time::Duration;
+
+use pn532::Interface;
+use pn532::{Pn532, Request};
+
+use core::task::Poll;
+
+use std::io::{Read, Write};
+
+use linux_embedded_hal as hal;
+
+const PROGRAM: Option<&'static str> = option_env!("CARGO_PKG_NAME");
+const VERSION: Option<&'static str> = option_env!("CARGO_PKG_VERSION");
+const DESCRIPTION: Option<&'static str> = option_env!("CARGO_PKG_DESCRIPTION");
+
+struct SerialPortInterface {
+    pub port: Box<dyn serialport::SerialPort>,
+}
+
+impl Interface for SerialPortInterface {
+    type Error = std::io::Error;
+
+    fn write(&mut self, frame: &[u8]) -> Result<(), Self::Error> {
+        match self.port.write(frame) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn wait_ready(&mut self) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
```

Comment:
Would it make sense to query if `self.port.bytes_to_read() > 0` and only then return `Poll::Ready`?